geoindicators/src/main/groovy/org/orbisgis/RsuIndicators.groovy

```diff
@@ -526,3 +526,97 @@ static IProcess rsuRoofAreaDistribution() {
                 [outputTableName: outputTableName]
             }
     )}
+
+
+/**
+ * Script to compute the effective terrain roughness (z0). The method for z0 calculation is based on the
+ * Hanna and Britter (2010) procedure (see equation (17) and examples of calculation p. 156 in the
+ * corresponding reference). It needs the "rsu_projected_facade_area_distribution" and the
+ * "rsu_geometric_mean_height" as input data.
+ *
+ * Warning: the calculation of z0 is only performed for angles included in the range [0, 180[°.
+ * To simplify the calculation, z0 is considered as equal for a given orientation independantly of the direction.
+ * This assumption is right when the RSU do not split buildings but could slightly overestimate the results
+ * otherwise (the real z0 is actually overestimated in one direction but OK in the opposite direction).
+ *
+ * References:
+ * Stewart, Ian D., and Tim R. Oke. "Local climate zones for urban temperature studies." Bulletin of the American Meteorological Society 93, no. 12 (2012): 1879-1900.
+ * Hanna, Steven R., and Rex E. Britter. Wind flow and vapor cloud dispersion at industrial and urban sites. Vol. 7. John Wiley & Sons, 2010.
+ *
+ * @param datasource A connexion to a database (H2GIS, PostGIS, ...) where are stored the input Table and in which
+ * the resulting database will be stored
+ * @param rsuTable the name of the input ITable where are stored the RSU geometries, the rsu_geometric_mean_height and
+ * the rsu_projected_facade_area_distribution fields
+ * @param projectedFacadeAreaName the name base used for naming the projected facade area field within the
+ * inputA rsuTable (default rsu_projected_facade_area_distribution - note that the field is also constructed
+ * with the direction and vertical height informations)
+ * @param geometricMeanBuildingHeightName the field name corresponding to the RSU geometric mean height of the
+ * roughness elements (buildings, trees, etc.) (in the inputA ITable)
+ * @param prefixName String use as prefix to name the output table
+ * @param listLayersBottom the list of height corresponding to the bottom of each vertical layers used for calculation
+ * of the rsu_projected_facade_area_density (default [0, 10, 20, 30, 40, 50])
+ * @param numberOfDirection the number of directions used for the calculation - according to the method used it should
+ * be divisor of 360 AND a multiple of 2 (default 12)
+ *
+ * @return outputTableName Table name in which the rsu id and their corresponding indicator value are stored
+ *
+ * @author Jérémy Bernard
+ */
+static IProcess rsuEffectiveTerrainRoughnessHeight() {
+    return processFactory.create(
+            "RSU effective terrain roughness height",
+            [rsuTable: String, projectedFacadeAreaName: String, geometricMeanBuildingHeightName: String,
+             prefixName: String, listLayersBottom: Double[], numberOfDirection: Integer,
```

Comment:
double

Reply:
The numberOfDirection must be an Integer. It is consistent with the rsuProjectedFacadeAreaDistribution IProcess

Reply:
Sorry I'm pointing listLayersBottom